operator/ingress/v1/types.go

```diff
@@ -0,0 +1,113 @@
+package v1
+
+import (
+	configv1 "github.com/openshift/api/config/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+
+// DNSRecord is a DNS record managed in the zones defined by
+// dns.config.openshift.io/cluster .spec.publicZone and .spec.privateZone.
+//
+// Cluster admin manipulation of this resource is not supported. This resource
+// is only for internal communication of OpenShift operators.
+type DNSRecord struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec is the specification of the desired behavior of the dnsecord.
+	Spec DNSRecordSpec `json:"spec,omitempty"`
+	// status is the most recently observed status of the dnsRecord.
+	Status DNSRecordStatus `json:"status,omitempty"`
+}
+
+// DNSRecordSpec contains the details of a DNS record.
+type DNSRecordSpec struct {
+	// dnsName is the hostname of the DNS record
+	//
+	// +kubebuilder:validation:Required
+	// +required
+	DNSName string `json:"dnsName,omitempty"`
+	// targets are record targets.
+	//
+	// +kubebuilder:validation:Required
+	// +required
+	Targets []string `json:"targets,omitempty"`
+	// recordType is the DNS record type. For example, "A" or "CNAME".
+	//
+	// +kubebuilder:validation:Required
+	// +required
+	RecordType DNSRecordType `json:"recordType,omitempty"`
+	// recordTTL is the record TTL in seconds. The default is 30.
+	//
+	// +kubebuilder:validation:Optional
+	// +optional
+	RecordTTL int64 `json:"recordTTL,omitempty"`
```

Comment:
is zero allowed?

Reply:
No, and it was never a pointer, so I'm going to make it required. Is there an openapi validation to enforce >=0?

Reply:
Looking at `controller-gen crd -ww`, I think you can use this marker:
```
+kubebuilder:validation:MinItems field
	=<int>  specifies the minimun length for this list.
```

Reply:
Fixed and added minimum validation